syntax = "proto3";

// Interface exported by the server.
service MavenCommandHandler {
  rpc RunCommand(Command) returns (Response) {}
}

// Data structure for the command input
message Command {
    string phase = 1;
    string pluginGoal = 2;
    string args = 3;
}

// Data structure for the response running the command
message Response {
    string output = 1;
    int32 exitCode = 2;
}

